databricks/koala/namespace.py

```diff
@@ -123,3 +126,59 @@ def _to_datetime2(arg_year, arg_month, arg_day,
         errors=errors,
         format=format,
         infer_datetime_format=infer_datetime_format)
+
+
+@derived_from(pd)
+def get_dummies(data, prefix=None, prefix_sep='_', dummy_na=False, columns=None, sparse=False,
+                drop_first=False, dtype=None):
+    if sparse is not False:
+        raise NotImplementedError("get_dummies currently does not support sparse")
+
+    if isinstance(columns, string_types):
+        columns = [columns]
+    if dtype is None:
+        dtype = 'byte'
+
+    if isinstance(data, Column):
+        if prefix is not None:
+            prefix = [str(prefix)]
+        columns = [data.name]
+        df = data.to_dataframe()
+        remaining_columns = []
+    else:
+        if isinstance(prefix, string_types):
+            raise ValueError("get_dummies currently does not support prefix as string types")
+        df = data.copy()
+        if columns is None:
+            columns = [column for column in df.columns
+                       if not isinstance(data.schema[column].dataType, NumericType)]
+        if prefix is None:
+            prefix = columns
+        remaining_columns = [df[column] for column in df.columns if column not in columns]
+
+    if prefix is not None and len(columns) != len(prefix):
+        raise ValueError(
+            "Length of 'prefix' ({}) did not match the length of the columns being encoded ({})."
+            .format(len(prefix), len(columns)))
+
+    all_values = df._spark_agg(*[F._spark_collect_set(F._spark_col(column)).alias(column)
```

Comment:
using `_unpack_scalar` in `stuctures.py` does not work?

Reply:
If there are issues of circular imports, we can move `_unpack_scalar` to utils.

Reply:
I see, you do multiple aggregations. Let's have a `_reduce_spark_multi` that takes a list of aggregators then.